src/types/gecko-profile.js

```diff
@@ -169,23 +169,17 @@ export type GeckoCounter = {|
   name: string,
   category: string,
   description: string,
-  sample_groups:
-    | {|
-        id: number,
-        samples: {|
-          schema: {|
-            time: 0,
-            number: 1,
-            count: 2,
-          |},
-          data: Array<[number, number, number]>,
-        |},
-      |}
-    // Due to a bug in Gecko, it's possible that we have a counter that has no
-    // sample data. This is likely a Gecko problem that we'll need to fix.
-    // See https://github.com/firefox-devtools/profiler/issues/2248 and
-    // https://bugzilla.mozilla.org/show_bug.cgi?id=1584190
-    | {||},
+  sample_groups: Array<{|
```

Comment:
nit: please use a `$ReadOnlyArray` instead of `Array` (if possible), so that Flow would guard against any mutable operation.

Reply:
Nice. But I don't see any usage of `$ReadOnlyArray` in that file for example and I think we can change most of them to that type. Do you think it's worth to an issue?

Reply:
yeah maybe :)